skimage/transform/_poly2mask.py

```diff
@@ -0,0 +1,53 @@
+import numpy as np
+import matplotlib
+
+from .. import draw
+
+
+def poly2mask(image_shape, polygon, backend='default'):
+    """Compute a mask from polygon.
+
+    Parameters
+    ----------
+    image_shape : tuple of size 2.
+        The shape of the mask.
+    polygon : array_like.
+        The polygon coordinates of shape (N, 2) where N is
+        the number of points.
+    backend : 'default' or 'matplotlib'.
+        The default implementation uses `skimage.draw`, while the
+        'matplotlib' implementtion uses `maplotlib.path.Path`.
+        While note exactly the same both methods give near-identical results.
+
+    Returns
+    -------
+    mask : 2-D ndarray of type 'bool'.
+        The mask that corresponds to the input polygon.
+    """
+    if backend == 'default':
+        return _poly2mask_skimage(image_shape, polygon)
+    elif backend == 'matplotlib':
+        return _poly2mask_mpl(image_shape, polygon)
+
+    raise ValueError(
+        "Wrong backend selected. Choose either 'default' or 'matplotlib'.")
+
+
+def _poly2mask_skimage(image_shape, polygon):
+    polygon = np.array(polygon)
+    fill_row_coords, fill_col_coords = draw.polygon(
+        polygon[:, 1], polygon[:, 0], image_shape)
+    mask = np.zeros(image_shape, dtype=np.bool)
+    mask[fill_row_coords, fill_col_coords] = True
+    return mask
+
+
+def _poly2mask_mpl(image_shape, polygon):
```

Comment:
Remove this backend. We are actually aiming to remove the mpl dependency for non-GUI code in the medium-term. At any rate we don't need two backends, it just adds to the complexity for little gain.